### PR TITLE
Create a script to do local builds of core Blockly.

### PR DESCRIPTION
**Pull request by rachel-fenichel**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Provides a script for doing local builds of Blockly, for people who are rate limited or don't have access to the closure compiler endpoint.

### Proposed Changes

The new script lives in `local_build/local_build.sh` and outputs to `local_build/local_blockly_compressed.js`.  I don't want to direct the output to `blockly_compressed.js` until I'm sure that the is equivalent.

### Reason for Changes

This is a common request.

### Test Coverage
Ran `./local_build.sh` and used the resulting file in the playground.  Played around to make sure things worked nicely.

Deleted `closure_compiler.jar` and confirmed that the appropriate error message showed up.

### Additional info
This doesn't fully match what `build.py` does.  One piece of low-hanging fruit is to correctly strip out the license text that `build.py` strips out.

### CODE REVIEW

File: local_build/local_build.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# through a local copy of Google's Closure Compiler with simple optimizations
+# turned on.
+
+# Future work:
+# - Trim down Google's Apache licenses, to match the output of build.py.
+# - Generate other compressed files generated by build.py normally.
+
+# Find the Closure Compiler.
+if [ -f "$(npm root)/google-closure-compiler/compiler.jar" ]; then
+  # Travis test.
```
Comment (AnmAtAnm) [inline local_build/local_build.sh]: Not specific to Travis.

File: local_build/local_build.sh [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+else
+  echo "ERROR: Closure Compiler not found."
+  echo "Download from this URL, and place jar file in current directory."
+  echo "https://dl.google.com/closure-compiler/compiler-latest.zip"
+  exit 1
+fi
+
+rm local_blockly_compressed.js 2> /dev/null
+echo Compiling Blockly core...
+java -jar closure-compiler.jar \
```
Comment (AnmAtAnm) [inline local_build/local_build.sh]: Use $COMPILER

File: local_build/local_build.sh [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+# Future work:
+# - Trim down Google's Apache licenses, to match the output of build.py.
+# - Generate other compressed files generated by build.py normally.
+
+# Find the Closure Compiler.
+if [ -f "$(npm root)/google-closure-compiler/compiler.jar" ]; then
+  # Travis test.
+  COMPILER="$(npm root)/google-closure-compiler/compiler.jar"
+elif [ -f *compiler*.jar ]; then
+  # Manual test.
```
Comment (AnmAtAnm) [inline local_build/local_build.sh]: Echo the path/filename of the compiler selected, here or below.
Add TODO about checking only a single file was found.

Add the same to tests/compile/compile.sh
(And while there, correct indentation on line 9 of that file.)

Comment (rachel-fenichel) [reply local_build/local_build.sh]: Done.

File: local_build/local_build.sh (hunk 1 above)
Comment (rachel-fenichel) [reply local_build/local_build.sh]: Removed.

**Comment by rachel-fenichel** [comment]
Tested in the playground.  The unit tests require uncompressed.

File: local_build/local_build.sh [hunk 4]
```diff
@@ -30,24 +30,26 @@
...
+  COMPILER="closure-compiler*.jar"
+  # TODO: Check whether multiple files were found.
 else
   echo "ERROR: Closure Compiler not found."
   echo "Download from this URL, and place jar file in current directory."
   echo "https://dl.google.com/closure-compiler/compiler-latest.zip"
   exit 1
 fi
 
+echo Using $COMPILER as the compiler.
```
Comment (AnmAtAnm) [inline local_build/local_build.sh]: Nit:
`echo "Using \"$COMPILER\" as the compiler."`

File: tests/compile/compile.sh [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
+  # TODO: Check whether multiple files were found.
 else
-	echo "ERROR: Closure Compiler not found."
+  echo "ERROR: Closure Compiler not found."
   echo "Download from this URL, and place jar file in current directory."
   echo "https://dl.google.com/closure-compiler/compiler-latest.zip"
   exit 1
 fi
 
+echo Using $COMPILER as the compiler.
```
Comment (AnmAtAnm) [inline tests/compile/compile.sh]: Same here.

**Pull request merged**